src/libsyntax/parse/parser.rs

```diff
@@ -2527,8 +2530,11 @@ impl<'a> Parser<'a> {
                 Err(mut e) => {
                     e.span_label(struct_sp, "while parsing this struct");
                     e.emit();
-                    self.recover_stmt();
-                    break;
+
+                    if self.token != token::Comma {
```

Comment:
This confuses me a bit — when does this `,` token get dropped, if not by `recover_stmt`?

Reply:
It gets dropped afterwards at line 2541 : 
```rust
self.expect_one_of(&[token::Comma], &[token::CloseDelim(token::Brace)])
```

Reply:
Ah, I see now. Yes, put a comment if you don't mind.